state-chain/pallets/cf-flip/src/on_charge_transaction.rs

```diff
@@ -74,11 +83,22 @@ impl<T: TxConfig + FlipConfig + Config> OnChargeTransaction<T> for FlipTransacti
 		_tip: Self::Balance,
 		escrow: Self::LiquidityInfo,
 	) -> Result<(), frame_support::unsigned::TransactionValidityError> {
-		if let Some(surplus) = escrow {
+		if let Some((surplus, call_info_id)) = escrow {
 			// It's possible the account was deleted during extrinsic execution. If this is the
 			// case, we shouldn't refund anything, we can just burn all fees in escrow.
 			let to_burn = if frame_system::Pallet::<T>::account_exists(who) {
-				corrected_fee
+				if let Some(call_info_id) = call_info_id {
+					let call_count = crate::CallCounter::<T>::mutate(&call_info_id, |count| {
+						*count += 1;
+						*count
+					});
+					match call_info_id {
+						CallInfoId::Pool { .. } =>
+							crate::FeeScalingRate::<T>::get().scale_fee(corrected_fee, call_count),
+					}
```

Comment:
The match statement is unnecessary? If there is *any* call info, we want to scale the fee according to the call count.